File: src/java/shared/src/main/java/com/aurora/shared/exception/DomainExceptions.java
package com.aurora.shared.exception;

/**
 * Domain Exception hierarchy matching C# DomainExceptions.cs
 */
public class DomainExceptions {

    public static class DomainException extends RuntimeException {
        public DomainException(String message) {
            super(message);
        }
        public DomainException(String message, Throwable cause) {
            super(message, cause);
        }
    }

    public static class NotFoundException extends DomainException {
        public NotFoundException(String message) {
            super(message);
        }
    }

    public static class ConflictException extends DomainException {
        public ConflictException(String message) {
            super(message);
        }
    }

    public static class ForbiddenException extends DomainException {
        public ForbiddenException(String message) {
            super(message);
        }
    }
}

File: src/dotnet/shared/Exceptions/DomainExceptions.cs
namespace Shared.Exceptions;

public class DomainException(string message) : Exception(message);
public class NotFoundException(string message) : Exception(message);
public class ConflictException(string message) : Exception(message);
public class ForbiddenException(string message) : Exception(message);
public class DomainValidationException(string message) : DomainException(message);
public class PolicyUnavailableException(string message) : Exception(message);
public class RiskPolicyNotConfiguredException(string message) : DomainException(message);

